src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs

```diff
@@ -589,10 +589,13 @@ private static HttpContext CloneHttpContext(HttpContext context)
             // The response is a dud, you can't do anything with it anyways
             var responseFeature = new HttpResponseFeature();
 
+            var endpointFeature = context.Features.Get<IEndpointFeature>();
+
             var features = new FeatureCollection();
             features.Set<IHttpRequestFeature>(requestFeature);
             features.Set<IHttpResponseFeature>(responseFeature);
             features.Set<IHttpConnectionFeature>(connectionFeature);
+            features.Set<IEndpointFeature>(endpointFeature);
```

Comment:
IMO you should create a basic implementation of IEndpointFeature, copy the endpoint across, and set that.

Reply:
For the other features we have a basic implementation available in the `Microsoft.AspNetCore.Http` package for exactly this purpose. Should we do something similar here? [`EndpointFeature`](https://github.com/aspnet/AspNetCore/blob/425c196cba530b161b120a57af8f1dd513b96f67/src/Http/Http.Abstractions/src/Routing/EndpointHttpContextExtensions.cs#L65) looks pretty dirt-simple and if we made it public it would basically be the same as whatever we could make here.

Reply:
I keep forgetting the `SetEndpoint` extension method exists 😕 

No I don't think we need to implement EndpointFeature because people can use `SetEndpoint` instead.

Speaking of which, @BrennanConroy instead of creating a basic endpoint class you could use `newHttpContext.SetEndpoint(endpoint)`